### PR TITLE
Clarify external CSR signerName field description

### LINKED ISSUE #98146: Clarify external CST signerName in docs and in CSR status fields
**What happened**:

I spent some time helping a customer figure out they were misusing the recently added `signerName` field in a CSR.  They had been using CSRs for several releases to generate server certs for Kubernetes services that were only used within the cluster.  They found and read the Signers section of the [Certificate Signing article](https://kubernetes.io/docs/reference/access-authn-authz/certificate-signing-requests/#signers).  The predefined Kubernetes signers didn't seem to fit their use case so they made up their own `signerName`.  When they approved the CSRs no certs were issued; it looked much as if the CSR controller was failing or not even running.

It was an area I was unfamiliar with as well and it took a long time to realize the signerName they were using was the cause of the behavior and this was actually working correctly.

The PR that added this function has a sentence that makes the intent of the signerName field clear:

> Add 'signerName' field to the CertificateSigningRequest API to allow for external signers to be built

Some ideas coming out of this exercise:

Could the signer code add a condition or an event for the CSR resource indicating that the control plane is not issuing a certificate for an approved CSR because the signerName represents an external signer?

Could the documentation on [Signers](https://kubernetes.io/docs/reference/access-authn-authz/certificate-signing-requests/#signers) clarify that other names are allowed and that those are external signers and as such the control plane signer will not issue certificates - that is function the administrator must provide or install ?

For people who have been using the v1beta1 API, would it be appropriate for the "Kubernetes signers" section to suggest that CSRs for internal Kubernetes services "can use kubernetes.io/kubelet-serving as described in [Manage TLS Certificates in a Cluster](https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster/#create-a-certificate-signing-request)"

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

- Create a CSR
- Send a CertificateSigningRequest resource to Kubernetes using:
```
cat <<EOF | kubectl apply -f -
apiVersion: certificates.k8s.io/v1
kind: CertificateSigningRequest
metadata:
  name: my-svc.my-namespace
spec:
  request: $(cat server.csr | base64 | tr -d '\n')
  signerName: test.io/test
  usages:
  - digital signature
  - key encipherment
  - server auth
EOF
```
- Approve the CSR: `kubectl certificate approve my-svc.my-namespace`
- CSR is Approved but not Issued:
```
$ kubectl get csr my-svc.my-namespace
NAME                  AGE   SIGNERNAME     REQUESTOR                    CONDITION
my-svc.my-namespace   95m   test.io/test   c02pbtnd0d1gporb83jg-admin   Approved
```

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):  I've repeated this on 1.18 through 1.20.1
- Cloud provider or hardware configuration: IBM Cloud Kubernetes Service
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by mengjiao-liu**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
Clarify external CSR signerName field description

#### Which issue(s) this PR fixes:

Fixes #98146

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/apis/certificates/types.go [hunk 1]
```diff
@@ -49,6 +49,12 @@ type CertificateSigningRequestSpec struct {
 	// `scope-hostname.io/name`.
 	// Distribution of trust for signers happens out of band.
 	// You can select on this field using `spec.signerName`.
+	// Kubernetes provides built-in signers that each have a well-known signerName:
+	// 1. kubernetes.io/kube-apiserver-client
+	// 2. kubernetes.io/kube-apiserver-client-kubelet
+	// 3. kubernetes.io/kubelet-serving
+	// 4. kubernetes.io/legacy-unknown
+	// Custom signerNames can also be specified and that those are external signers and as such the control plane signer will not issue certificates.
 	SignerName string
```
Comment (liggitt) [inline pkg/apis/certificates/types.go:58]: this is already documented on the v1 struct:

https://github.com/kubernetes/kubernetes/blob/184c6d348c80c1f529b742c8959677817689a182/staging/src/k8s.io/api/certificates/v1/types.go#L64-L85

I'd suggest copying the documentation from there if we want to also update this field's doc

Comment (mengjiao-liu) [reply pkg/apis/certificates/types.go:58]: Thanks. I will revise it

**Pull request merged**